src/rad/core.clj

```diff
@@ -6,22 +6,44 @@
 
 ;; point is x & y position of the point (or cursor)
 (def point (atom [0 0]))
+(declare sync-frontend-cursor-to-point-atom! move-point-backwards! move-point-backwards)
+
+(defn move-point-backwards
+  "Returns point 'steps' steps backwards until it hits a beginning-of-line"
+  [buffer point steps]
+  (let [can-move-backwards? (fn [point-x]
+                              (> point-x 0))]
+    (loop [line (buffer (second point))
+           point-x (first point)
+           steps steps]
+      (if (and
+           (> steps 0)
+           (can-move-backwards? point-x))
+
+        (recur line (dec point-x) (dec steps))
+        [point-x (second point)]))))
+
+(defn move-point-backwards!
+  "Moves point 'steps' steps backwards, and syncs the UI cursor"
+  [steps] (do
+            (reset! point (move-point-backwards @rad.buffer/current-buffer @point steps))
+            (sync-frontend-cursor-to-point-atom!)))
 
-;; FIXME understand newlines
 (defn move-point-forward
   "Returns point 'steps' steps forward until it hits a newline."
-  [buffer point steps] (let [line (buffer (second point))
-                             point-x (first point)
-                             point-y (second point)
-                             can-move-forward? (fn [line p-x]
-                                                 (not (>= p-x (count line))))]
+  [buffer point steps]
+  (let [line (buffer (second point))
+        point-x (first point)
+        point-y (second point)
+        can-move-forward? (fn [line p-x]
+                            (not (>= p-x (count line))))]
 
-                         (if (can-move-forward? line point-x)
-                           (recur
-                            buffer
-                            [(+ 1 point-x) point-y] ;; one step forward
-                            (- steps 1))
-                           point)))
+    (if (and (> steps 0) (can-move-forward? line point-x))
```

Comment:
Too much in one line. Put the two sexprs inside `(and )` on new lines. Readability yo